app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php

```diff
@@ -0,0 +1,135 @@
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Catalog\Test\Unit\Observer;
+
+use Magento\Framework\Event;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Framework\Stdlib\DateTime\Timezone;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Observer\SetSpecialPriceStartDate;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Unit test for \Magento\Catalog\Observer\SetSpecialPriceStartDate
+ */
+class SetSpecialPriceStartDateTest extends TestCase
+{
+    /**
+     * Testable Object
+     *
+     * @var SetSpecialPriceStartDate
+     */
+    private $observer;
+
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @var Observer|MockObject
+     */
+    private $observerMock;
+
+    /**
+     * @var Event|MockObject
+     */
+    private $eventMock;
+
+    /**
+     * @var Product|MockObject
+     */
+    private $productMock;
+
+    /**
+     * @var Timezone|MockObject
+     */
+    private $timezone;
+
+    /**
+     * @var \DateTime|MockObject
+     */
+    private $dateObject;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp() : void
+    {
+        $this->objectManager = new ObjectManager($this);
+        $this->observerMock = $this->createMock(Observer::class);
+        $this->timezone = $this->createMock(Timezone::class);
+        $this->dateObject = $this->createMock(\DateTime::class);
+
+        $this->eventMock = $this->getMockBuilder(Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getProduct'])
+            ->getMock();
+
+        $this->productMock = $this->getMockBuilder(Product::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getSpecialPrice', 'getSpecialFromDate', 'setData'])
+            ->getMock();
+
+        $this->observer = $this->objectManager->getObject(
+            SetSpecialPriceStartDate::class,
+            [
+                'localeDate' => $this->timezone
+            ]
+        );
+    }
+
+    /**
+     * Test observer execute method
+     */
+    public function testExecuteModifySpecialFromDate()
+    {
+        $specialPrice = 15;
+        $specialFromDate = null;
+        $localeDateMock = ['special_from_date' => $this->returnValue($this->dateObject)];
+
+        $this->observerMock
+            ->expects($this->once())
+            ->method('getEvent')
+            ->willReturn($this->eventMock);
+
+        $this->eventMock
+            ->expects($this->once())
+            ->method('getProduct')
+            ->willReturn($this->productMock);
+
+        $this->dateObject->expects($this->any())
+            ->method('setTime')
+            ->will($this->returnSelf());
+
+        $this->timezone
+            ->expects($this->once())
+            ->method('date')
+            ->will($this->returnValue($this->dateObject));
```

Comment:
I would recommend using `willReturn` for better consistency with other stubs.

```suggestion
            ->willReturn($this->dateObject));
```

Reply:
@dmytro-ch fixed, please check. Tnx!